File: lcdIndividual/LCDAnimation/LCDAnimation/ViewController.m
//
//  ViewController.m
//  LCDAnimation
//
//  Created by ReleasePackageMachine on 2017/12/26.
//  Copyright © 2017年 com.1ktower.snakepop. All rights reserved.
//

#import "ViewController.h"
#import "LCDAnimation.h"
#import "Gradient.h"
#import "BezierPath.h"
#import "LiziAnimation.h"
@interface ViewController ()
@property (nonatomic,strong)CALayer * myLayer;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    LiziAnimation * ani = [[LiziAnimation alloc] init];
    ani.frame = self.view.bounds;
    [self.view addSubview:ani];
    
    [self.view.layer addSublayer:self.myLayer];
    self.myLayer.contents = (__bridge id _Nullable)([UIImage imageNamed:@"1024"].CGImage);
    NSLog(@"%@",[UIImage imageNamed:@"1024"]);
    self.myLayer.shadowColor = [UIColor grayColor].CGColor;
    self.myLayer.shadowOffset = CGSizeMake(50, 10);
    self.myLayer.shadowOpacity = 0.6;
    self.myLayer.cornerRadius = 100;
    self.myLayer.masksToBounds = YES;
    self.myLayer.borderColor = [UIColor brownColor].CGColor;
    self.myLayer.borderWidth = 5.0;
    // Do any additional setup after loading the view, typically from a nib.
}

-(CALayer *)myLayer{
    if (!_myLayer) {
        _myLayer = [CALayer layer];
        UIImage *img = [UIImage imageNamed:@"1024"];
        _myLayer.frame = CGRectMake(0, 468, 200, 200);
        _myLayer.backgroundColor = [UIColor greenColor].CGColor;
    }
    return _myLayer;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
@end

File: lcdIndividual/ReSideMenu/Myfile/ViewController.swift
//
//  ViewController.swift
//  ReSideMenu
//
//  Created by ReleasePackageMachine on 2017/12/28.
//  Copyright © 2017年 com.1ktower.snakepop. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    let sideVC = SideMenuViewController()
    var count = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()
        let width = UIScreen.main.bounds.size.width;
        let height = UIScreen.main.bounds.size.height;
        
        self.navigationItem.leftBarButtonItem = UIBarButtonItem(title: "Left", style: .plain, target: self, action: #selector(self.onLeftClicked))
        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Right", style: .plain, target: self, action: #selector(self.onRightClicked))
        self.title = "Home"
        self.view.backgroundColor = UIColor.green

        sideVC.view.frame = CGRect(x:-width,y:0,width:width,height:height)
        self.addChildViewController(sideVC)
        self.view.addSubview(sideVC.view)
        self.view.bringSubview(toFront: sideVC.view)
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    @objc func onLeftClicked(sender: UIBarButtonItem) {
        count = count + 1
        let width = UIScreen.main.bounds.size.width;
        let height = UIScreen.main.bounds.size.height;
        weak var weakSelf = self
        if count%2 != 0 {
            UIView.animate(withDuration: 0.2, animations: {
                weakSelf?.navigationController?.view.transform = CGAffineTransform.init(translationX: width/2, y: 0)
                
//                weakSelf?.view.center = CGPoint(x:width/2,y:height/2)
//                weakSelf?.view.bringSubview(toFront: (weakSelf?.sideVC.view!)!)
            })
        }else{
            UIView.animate(withDuration: 0.2, animations: {
                weakSelf?.navigationController?.view.transform = CGAffineTransform.identity
//                weakSelf?.view.center = CGPoint(x:0,y:height/2)
            })
            
        }
        
        
        print("left")
    }
    @objc func onRightClicked() {
        count = count + 1
        let width = UIScreen.main.bounds.size.width;
        let height = UIScreen.main.bounds.size.height;
        weak var weakSelf = self
        if count%2 != 0 {
            UIView.animate(withDuration: 0.2, animations: {
                weakSelf?.navigationController?.view.transform = CGAffineTransform.init(translationX: -width/2, y: 0)
                //                weakSelf?.view.center = CGPoint(x:width/2,y:height/2)
                //                weakSelf?.view.bringSubview(toFront: (weakSelf?.sideVC.view!)!)
            })
        }else{
            UIView.animate(withDuration: 0.2, animations: {
                weakSelf?.navigationController?.view.transform = CGAffineTransform.identity
                //                weakSelf?.view.center = CGPoint(x:0,y:height/2)
            })
            
        }
        print("right")
    }

}

